src/System.Reflection/tests/FieldInfo/FieldInfo_MethodTests.cs

```diff
@@ -86,197 +39,123 @@ public void TestGetValue_Exception1()
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (int)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((int)fi.GetValue(myInstance)).Equals(newfieldValue));
-        }
-
-        //Verify SetValue method for static String FieldType
-        [Fact]
-        public void TestSetValue_StaticStrField()
-        {
-            String fieldname = "static_strField";
-            String origfieldvalue;
-            String newfieldValue = "new";
-            FieldInfo fi = GetField(fieldname);
-            FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
-
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
-
-            origfieldvalue = (String)fi.GetValue(myInstance);
-            fi.SetValue(myInstance, (object)newfieldValue);
-
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(newfieldValue));
-
-            //reset static field value to its original value
-            FieldInfoMethodTests.static_strField = origfieldvalue;
+            FieldInfo fi = GetAndTestFieldInfo("nonstatic_strField");
+            object objInstance = new object();
+            Assert.Throws<ArgumentException>(() => { string retValue = (string)fi.GetValue(objInstance); });
         }
 
-        //Verify SetValue method for non-static String FieldType
-        [Fact]
-        public void TestSetValue_StrField()
+        // Verify SetValue method for fields
+        [Theory]
+        [InlineData("intFieldStatic", 1000)]
+        [InlineData("intFieldNonStatic", 1000)]
+        [InlineData("static_strField", "new")]
+        [InlineData("nonstatic_strField", "new")]
+        public void TestSetValue(string fieldName, object newFieldValue)
         {
-            String fieldname = "nonstatic_strField";
-            String newfieldValue = "new";
-            FieldInfo fi = GetField(fieldname);
+            FieldInfo fi = GetAndTestFieldInfo(fieldName);
             FieldInfoMethodTests myInstance = new FieldInfoMethodTests();
 
-            Assert.NotNull(fi);
-            Assert.True(fi.Name.Equals(fieldname));
+            object origFieldValue = fi.GetValue(myInstance);
+            fi.SetValue(myInstance, newFieldValue);
+            Assert.Equal(newFieldValue, fi.GetValue(myInstance));
 
-            fi.SetValue(myInstance, (object)newfieldValue);
-            Assert.True(((String)fi.GetValue(myInstance)).Equals(newfieldValue));
+            //reset static field values to their original value
+            switch (fieldName)
+            {
+                case "intFieldStatic":
+                    intFieldStatic = (int)origFieldValue;
+                    break;
+                case "static_strField":
+                    static_strField = (string)origFieldValue;
+                    break;
+            }
```

Comment:
We could instead use fi.SetValue to do this, both to reset the value and as secondary validation, e.g.
```C#
public void SetValue_Roundtrip_InstanceAndStaticFields(string fieldName, object newFieldValue)
{
    FieldInfo fi = GetAndTestFieldInfo(fieldName);
    var myInstance = new FieldInfoMethodTests();
    object origFieldValue = fi.GetValue(myInstance);

    fi.SetValue(myInstance, newFieldValue);
    Assert.Equal(newFieldValue, fi.GetValue(myInstance));

    fi.SetValue(myInstance, origFieldValue);
    Assert.Equal(origFieldValue, fi.GetValue(myInstance));
}
```